src/GitHub.App/Controllers/NavigationController.cs

```diff
@@ -66,7 +66,15 @@ public void LoadView(IConnection connection, ViewWithData data, Action<IView> on
             switch (data.MainFlow)
             {
                 case UIControllerFlow.PullRequestCreation:
+                    //The selected flow will change after the view is created, so save it.
+                    var previousFlow = Current?.SelectedFlow;
+
                     CreateView(connection, data, onViewLoad);
+                    if (data.Data == null && previousFlow == UIControllerFlow.PullRequestCreation)
+                    {
+                        Pop(history[history.Count-2]);
```

Comment:
This line of code makes the assumption that `history.Count` is always two or more if you make it inside this `if` block. That's not immediately clear to a drive-by reviewer. These are the kinds of cases where if you're sure that is always true, document that assumption via a `Debug.Assert`.

Something like `Debug.Assert(history.Count >= 2, "Uh oh, History count was less than two but that's impossible if the previous flow is PullRequestCreation");`